### PR TITLE
Added, to the DataStream API guide, an example of how to collect results.

### PR DESCRIPTION
**Pull request by filipefigcorreia**
This is similar to what already exists for the DataSet API guide: 
https://ci.apache.org/projects/flink/flink-docs-release-0.10/apis/programming_guide.html#collection-data-sources-and-sinks

### CODE REVIEW

File: docs/apis/streaming_guide.md [hunk 1]
```diff
@@ -1830,6 +1840,17 @@ val myTuples = env.fromCollection(data)
...
 {% endhighlight %}
+
+A collection data sink is specified as follows:
+
+{% highlight scala %}
+import org.apache.flink.contrib.streaming.DataStreamUtils
+import scala.collection.JavaConverters.asScalaIteratorConverter
+
+val myResult: DataStream[(String, Int)] = ...
+val myOutput = DataStreamUtils.collect(myResult.getJavaStream).asScala.toList
```
Comment (rmetzger) [inline docs/apis/streaming_guide.md]: does the `toList()` call work on infinite streams?

**Comment by fhueske** [comment]
Thanks for the pull request! This is valuable information for testing and debugging DataStream programs and should be documented. 

However, the sink does not return a collection but an iterator. I think it would be better to move this to a separate section called "Iterator Data Sink" which follows the "Collection Data Sources" section. We should also add that this sink should only be used for testing and remove the `toList` call as suggested by @rmetzger.

**Comment by filipefigcorreia** [comment]
Sounds good. Would you say that we should update the DataSet API guide to match this structure? We currently have a "Collection Data Sources and Sinks" section that mentions a DataSink that may also only be used locally, for debugging purposes. That one doesn't return an iterator though.

**Comment by fhueske** [comment]
Thanks for the fast update! Will merge this PR.

In DataSet the sink actually returns a collection. I think it is fine to keep it as it is.

**Pull request closed without merging**